Src/GoogleApis.Auth/OAuth2/ClientCredentialParameters.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+Copyright 2015 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Google.Apis.Json;
+using Google.Apis.Util;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>Indicates the type of Client Credential.</summary>
+    public enum ClientCredentialType
+    {
+        /// <summary>UserCredential is created by the gcloud sdk tool when the user runs <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        UserCredential = 1,
+        /// <summary>ServiceAccountCredential is downloaded by the user from <a href="https://console.developers.google.com">Google Developers Console</a>.</summary>
+        ServiceAccountCredential = 2
+    }
+
+    /// <summary>Container for the union of parameters/properties from all credential types.</summary>
+    public sealed class ClientCredentialParameters
```

Comment:
It's funny because I actually didn't offer the name (or at least I don't remember that I did), and I just thought to comment that I don't really like it, and then I saw this conversation...
:)

All other "json" classes that we have are pretty straight forward,
TokenResponse
ClientSecret
TokenRequest
...

I'm not sure what ClientCredentialParameters implies for?
I also find it weird that there is a fromJsonKey method that gets ClientCredentialParameters....